process/factory/interceptorscontainer/shardInterceptorsContainerFactory.go

```diff
@@ -0,0 +1,242 @@
+package interceptorscontainer
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/core/throttler"
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/process/factory/containers"
+	interceptorFactory "github.com/ElrondNetwork/elrond-go/process/interceptors/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+var _ process.InterceptorsContainerFactory = (*shardInterceptorsContainerFactory)(nil)
+
+// shardInterceptorsContainerFactory will handle the creation the interceptors container for shards
+type shardInterceptorsContainerFactory struct {
+	*baseInterceptorsContainerFactory
+	keyGen        crypto.KeyGenerator
+	singleSigner  crypto.SingleSigner
+	addrConverter state.AddressConverter
+}
+
+// NewShardInterceptorsContainerFactory is responsible for creating a new interceptors factory object
+func NewShardInterceptorsContainerFactory(
+	accounts state.AccountsAdapter,
```

Comment:
an arg would reduce the dupplicate parameters

Reply:
done